log/log.go

```diff
@@ -121,27 +210,57 @@ func (hook ContextHook) Levels() []logrus.Level {
 	return logrus.AllLevels
 }
 
+func fmtCaller(frame runtime.Frame) string {
+	return fmt.Sprintf(
+		"%s:%d>%s",
+		path.Base(frame.File),
+		frame.Line,
+		path.Base(frame.Function),
+	)
+}
+
 func (hook ContextHook) Fire(entry *logrus.Entry) error {
-	//'skip' = 6 is the default call stack skip, which
-	//works ootb when Error(), Warn(), etc. are called
-	//for Errorf(), Warnf(), etc. - we have to skip 1 lvl up
-	for skip := 6; skip < 8; skip++ {
-		if pc, file, line, ok := runtime.Caller(skip); ok {
-			funcName := runtime.FuncForPC(pc).Name()
-
-			//detect if we're still in logrus (formatting funcs)
-			if !strings.Contains(funcName, "github.com/sirupsen/logrus") {
-				entry.Data["file"] = path.Base(file)
-				entry.Data["func"] = path.Base(funcName)
-				entry.Data["line"] = line
```

Comment:
ok you have to educate me a bit here.  we used those, as they are well-defined consts from logrus (e.g.:FieldKeyFunc). now we are only using "caller" which is not a well defined const and is unused in logrus code. how does it work?

Reply:
I wasn't aware that these were "well-defined logrus consts". It just seems pointless to index the line number, filename and function independently in the log parser (or indexing the caller context at all seems pretty pointless), so I condensed it to a single field.